x/stake/keeper.go

```diff
@@ -710,6 +728,15 @@ func (k Keeper) Validator(ctx sdk.Context, addr sdk.Address) sdk.Validator {
 	return val
 }
 
+// get the sdk.validator for a particular pubkey
+func (k Keeper) ValidatorByPubKey(ctx sdk.Context, pubkey crypto.PubKey) sdk.Validator {
```

Comment:
Let's delete this function, it just confuses things - functions should just refer to `GetValidatorByPubKey` and check if found or not

Reply:
(and this function is only referenced in the keeper interface, and in tests)

Reply:
Agreed, deleted.